### PR TITLE
set extracted blood temp

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary
Bugfixes "Set extracted blood temperature"

#### Purpose of change
Fix: #48971

#### Describe the solution

Set the temperature of the extracted blood (or acid) item.
Blood from corpses is set to corpse temperature.
Blood from yourself is at 37C,

#### Testing

Extract blood from corpse/yourself. The blood is warm.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Shouldn't the temperature of blood taken from the PC (or any other living creature) be affected by their actual body temperature (fever, hypothermia, body temperature affecting mutations, species...)?

Regardless, a default value is much better than an undefined one.

**Comment by Hirmuolio** [comment]
If someone wants to define body temperatures for injuries/mutations we can use those.
But until that happens 37 C is good enough. Few C difference is practically insignificant for item temperatures.

File: src/iuse.cpp [hunk 1]
```diff
@@ -4751,6 +4753,7 @@ cata::optional<int> iuse::blood_draw( player *p, item *it, bool, const tripoint
     if( !drew_blood && query_yn( _( "Draw your own blood?" ) ) ) {
         p->add_msg_if_player( m_info, _( "You drew your own blood…" ) );
         drew_blood = true;
+        blood_temp = 310.15;
```
Comment (BrettDong) [inline src/iuse.cpp:4756]:
```suggestion
        blood_temp = 310.15f;
```

File: src/iuse.cpp [hunk 2]
```diff
@@ -4733,12 +4733,14 @@ cata::optional<int> iuse::blood_draw( player *p, item *it, bool, const tripoint
     item blood( "blood", calendar::turn );
     bool drew_blood = false;
     bool acid_blood = false;
+    float blood_temp;  //kelvins
```
Comment (BrettDong) [inline src/iuse.cpp:4736]:
```suggestion
    float blood_temp = 0.0f;  //kelvins
```

Comment (PatrikLundell) [reply src/iuse.cpp:4736]: Why that default value?

The negative illegal indicator one mentioned in the referenced issue report would make some kind of sense, as would one that had some somewhat realistic "average reasonable" value (such as anywhere in the range 0 to 25 degrees Celsius). A default value of 0 degrees Kelvin is always going to be wrong, with the only advantage being that it probably doesn't cause any code to crash to indicate the result is nonsensical (unless you try to divide by temperature, but I can't currently see why you'd try to do that).

Comment (BrettDong) [reply src/iuse.cpp:4736]: > Why that default value?

In order to silence the compile warning.

**Comment by Fris0uman** [comment]
Why close it?

**Pull request merged**